core/src/validator.rs

```diff
@@ -168,7 +170,44 @@ impl Validator {
             config.storage_slots_per_turn,
             bank.slots_per_segment(),
         );
-
+        let bf = bank_forks.clone();
+        let kp = keypair.clone();
```

Comment:
you could just use local variable `id`, declared higher up in this function.  The validator's public key (aka, `id`) will never change at runtime.